src/main/scala/viper/gobra/frontend/info/implementation/resolution/NameResolution.scala

```diff
@@ -283,10 +288,21 @@ trait NameResolution { this: TypeInfoImpl =>
         } else lookup(sequentialDefenv(n), serialize(n), UnknownEntity()) // otherwise it is just a variable
 
       case n =>
-        (n, lookup(sequentialDefenv(n), serialize(n), UnknownEntity())) match {
+        (n, lookup(sequentialDefenv(n), serialize(n), UnknownEntity()), tryEnclosingInterface(n)) match {
```

Comment:
the third argument of lookup is lazy. you could put the computation that happens if lookup fails in there.

Reply:
This was actually even buggy. I changed it now to first perform a lookup in unqualified imports (level 2) and only then in the interface (level 3). The interface lookup happens only if level 1 lookup has failed but I do it together with the level 2 lookup to save a few indentation levels. If you think this is not nice or causes performance issues I'm happy to change it to a truly serial way